api/circulation.py

```diff
@@ -717,3 +717,64 @@ def default_notification_email_address(self, patron, pin):
         of changes?
         """
         return Configuration.default_notification_email_address()
+
+
+    def checkin(self, patron, pin, licensepool):
+        """  Return a book early.  
+
+        :param patron: a Patron object for the patron who wants
+        to check out the book.
+        :param pin: The patron's alleged password.
+        :param licensepool: Identifier of the book to be checked out is 
+        attached to this licensepool.
```

Comment:
It's not just the identifier -- the _LicensePool_ is the book that is going to be checked out. Right now there is only one LicensePool per Identifier, but that's going to change pretty soon.